bot.py

```diff
@@ -16,34 +17,130 @@ class MusicBot(discord.Client):
     The main bot functionality
     """
 
+    COMMAND_PREFIX = "!"
+
+    def __init__(self):
+        self.handlers = {}
+
+        self.register_command("hello", handler=self.hello)
+        self.register_command("countdown", handler=self.countdown)
+        self.register_command("dinkster", handler=self.dinkster)
+
+        super().__init__()
+
+    def register_command(self, command_name, handler=None):
+        """Register a command with the name 'command_name'.
+
+        Arguments:
+          command_name: String. The name of the command to register. This is
+            what users should use to run the command.
+          handler: A function. This function should accept two arguments:
+            discord.Message and a string. The messageris the message being
+            processed by the handler and command_content is the string
+            contents of the command passed by the user.
+        """
+        assert handler
+        assert command_name not in self.handlers
+        self.handlers[command_name] = handler
+
+    def get_command_handler(self, message_content):
+        """Tries to parse message_content as a command and fetch the corresponding handler for the command.
+
+        Arguments:
+          message_content: Contents of the message to parse. Assumed to start with COMMAND_PREFIX.
+
+        Returns:
+          handler: Function to handle the command.
+          command_content: The message contents with the prefix and command named stripped.
+          error_msg: String or None. If not None it signals that the command was
+            unknown. The value will be an error message displayable to the user which
+            says the command was not recognized.
+        """
+        if message_content[0] != "!":
```

Comment:
`"!"` should be `MusicBot.COMMAND_PREFIX`
However, this seems redundant anyway since COMMAND_PREFIX is already checked earlier in the call stack (line 116).